test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -709,10 +740,21 @@ public void AnomalyDetectionWithSrCnn(bool loadDataFromFile)
         {
             var ml = new MLContext(1);
             IDataView dataView;
+            var data = new List<TimeSeriesDataDouble>();
             var dataPath = GetDataPath("Timeseries", "period_anomaly.csv");
 
             // Load data from file into the dataView
-            dataView = ml.Data.LoadFromTextFile<TimeSeriesDataDouble>(dataPath, hasHeader: true);
+            StreamReader readFile = new StreamReader(dataPath);
+            string line;
```

Comment:
Any particular reason LoadFromTextFile doesn't work for you?

Reply:
+1, `LoadFromTextFile` is meant for this exact purpose of loading from a data path with a class type provided. Is there a specific reason why you don't use `LoadFromTextFile`?

Reply:
Basically, I want to enumerate the input data in the test to verify the input value is bounded by the output value 'upperBound' and 'lowerBound'. Could you point me an example that I can use the DataView to do this? Thanks.

---
In reply to: [482242678](https://github.com/dotnet/machinelearning/pull/5374#discussion_r482242678) [](ancestors = 482242678)

Reply:
You can use `CreateEnumerable`  to convert an IDataView into an enumerable. See [https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.dataoperationscatalog.createenumerable?view=ml-dotnet](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.dataoperationscatalog.createenumerable?view=ml-dotnet)

---
In reply to: [482675482](https://github.com/dotnet/machinelearning/pull/5374#discussion_r482675482) [](ancestors = 482675482,482242678)

Reply:
Updated.

---
In reply to: [482718920](https://github.com/dotnet/machinelearning/pull/5374#discussion_r482718920) [](ancestors = 482718920,482675482,482242678)